projects/packages/jetpack-mu-wpcom/src/features/replace-site-visibility/site-visibility/index.tsx

```diff
@@ -0,0 +1,244 @@
+import { sprintf, __ } from '@wordpress/i18n';
+import { useState } from 'react';
+import SitePreviewLink from '../site-preview-link';
+import type { SitePreviewLinkObject } from '../site-preview-link';
+import './style.scss';
+
+interface Props {
+	homeUrl: string;
+	isWpcomStagingSite: boolean;
+	isUnlaunchedSite: boolean;
+	hasSitePreviewLink: boolean;
+	sitePreviewLink?: SitePreviewLinkObject;
+	sitePreviewLinkNonce: string;
+	blogPublic: number;
+	wpcomComingSoon: number;
+	wpcomPublicComingSoon: number;
+	wpcomDataSharingOptOut: boolean;
+}
+
+const SiteVisibility = ( {
+	homeUrl,
+	isWpcomStagingSite,
+	isUnlaunchedSite,
+	hasSitePreviewLink,
+	sitePreviewLink,
+	sitePreviewLinkNonce,
+	blogPublic: defaultBlogPublic,
+	wpcomComingSoon: defaultWpcomComingSoon,
+	wpcomPublicComingSoon: defaultWpcomPublicComingSoon,
+	wpcomDataSharingOptOut: defaultWpcomDataSharingOptOut,
+}: Props ) => {
+	const [ fields, updateFields ] = useState( {
+		blogPublic: Number( defaultBlogPublic ),
+		wpcomComingSoon: Number( defaultWpcomComingSoon ) === 1,
+		wpcomPublicComingSoon: Number( defaultWpcomPublicComingSoon ) === 1,
+		wpcomDataSharingOptOut: defaultWpcomDataSharingOptOut,
+	} );
+
+	const { blogPublic, wpcomComingSoon, wpcomPublicComingSoon, wpcomDataSharingOptOut } = fields;
+
+	// isPrivateAndUnlaunched means it is an unlaunched coming soon v1 site
+	const isPrivateAndUnlaunched = -1 === blogPublic && isUnlaunchedSite;
+	const isAnyComingSoonEnabled =
+		( 0 === blogPublic && wpcomPublicComingSoon ) || isPrivateAndUnlaunched || wpcomComingSoon;
+	const isPublicChecked = ( blogPublic === 0 && ! wpcomPublicComingSoon ) || blogPublic === 1;
+	const showPreviewLink =
+		Number( defaultWpcomPublicComingSoon ) === 1 && isAnyComingSoonEnabled && hasSitePreviewLink;
+	const discourageSearchChecked = 0 === blogPublic && ! wpcomPublicComingSoon;
+
+	return (
+		<>
+			<p>
```

Comment:
I think we should add `description` class here to make it gray, similar to the Related posts setting:

<img width="991" alt="image" src="https://github.com/user-attachments/assets/0d6a497b-61f5-4d4b-97be-898216efd473" />

Reply:
The reason I didn't add `description` class is that it's the main content of the Site Visibility field and its color on Calypso is also different from the description of other settings. I don't have strong opinion on it so I'll add the `description` class to align with the Related posts setting.

Reply:
> I don't have strong opinion on it so I'll add the description class to align with the Related posts setting.

Did we add it 🙈

Reply:
Done by [e89f23b](https://github.com/Automattic/jetpack/pull/42230/commits/e89f23b954f0868295c9ab72cd18f931c55a80d8).